src/alire/alire-toolchains.adb

```diff
@@ -487,6 +487,17 @@ package body Alire.Toolchains is
             end if;
          end;
       end if;
+
+      --  Remove caching of external condition too for consistency
+
+      if CLIC.Config.Defined (Config.DB.all, Tool_Key (Crate, For_Is_External))
+      then
+         Trace.Debug
+           ("Unsetting " & Tool_Key (Crate, For_Is_External) & ": "
+            & CLIC.Config.Edit.Unset
+              (Config.Edit.Filepath (Level),
+               Tool_Key (Crate, For_Is_External))'Image);
+      end if;
```

Comment:
This is a bugfix for a corner case when the user selected None in the assistant after having selected a regular compiler, causing detection of the external compiler to be skipped. It was exposed when tweaking a test to remove `--uninstall`.